### PR TITLE
Refactor monthly history view to use retrospective 12-month window

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context
**Type:** Refactor  
**Intent:** Change the monthly history view to produce a retrospective 12-month window that ends at the selected month (selected month is the rightmost bar). Fetches and merges 12 months of current-period data and the corresponding 12 months from the previous year to enable consistent year-over-year comparisons in the chart.  

### 🧭 Reviewer Guide
**Complexity:** Medium

#### Entry Point
Start with `components/features/history/MonthlyHistoryGraph.tsx` — this contains the core refactor: month parsing, calculation of the 12-month retrospective window, two API fetches (current period and previous-year window), map-based merging logic, and construction of the 12-entry chart payload. Then review `components/features/history/MonthlyHistoryChart.tsx` to see the client-side rendering changes (the `ConditionalBarShape` that prevents zero-value bars from rendering and the Bar components using it). Finally, inspect `components/features/history/MonthlyHistoryGraph.test.tsx` to verify expected behaviors and edge cases (12-month length, zero-month handling, tag propagation, startMonth calculations, and failure handling).

#### Sensitive Areas
- `components/features/history/MonthlyHistoryGraph.tsx` - month arithmetic (parse/subMonths/subYears), startMonth calculations, year-boundary logic, and the map-based merge producing the 12-month ordered dataset.  
- `components/features/history/MonthlyHistoryChart.tsx` - `ConditionalBarShape` rendering behavior and use of `BarRectangleItem`; ensure tooltip/legend formatting remains correct.  
- `components/features/history/MonthlyHistoryGraph.test.tsx` - confirm tests correctly mock fetches and assert merged output and API call parameters.  
- `components/features/history/MonthlyHistoryRow.tsx` - changed date formatting to `date-fns` `format(payment.paidAt, 'MMM d, yyyy')` (presentation change to verify).  
- `scripts/seed-development.ts` - seeding logic now uses `now` as reference and filters future-dated generated transactions (affects development data).

### ⚠️ Risk Assessment
- **Breaking Changes:** No breaking changes - internal logic, rendering, tests, documentation, and seed script updated; no public API or component interface changes.  
- **Migrations/State:** No migrations or state changes

### CODE REVIEW

File: components/features/history/MonthlyHistoryRow.tsx [hunk 1]
```diff
@@ -35,13 +36,7 @@ export function MonthlyHistoryRow({
...
       </td>
       <td className="text-right">
-        <span>
-          {new Intl.DateTimeFormat(locale, {
-            day: 'numeric',
-            month: 'long',
-            year: 'numeric',
-          }).format(payment.paidAt)}
-        </span>
+        <span>{format(payment.paidAt, 'MMM d, yyyy')}</span>
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/history/MonthlyHistoryRow.tsx]: We DO NOT use locales yet for dates  on other pages. Why do force me to introduce entropy?
